### DISCUSSION TITLE
myUrl/databases/myDatabase/indexes/progress via c# Store

### DISCUSSION THREAD

**Discussion by stevencasburn** [ravendb/ravendb#16263] [Q&A] [answered]
Hi,

I've found that I can request `myUrl/databases/myDatabase/indexes` by doing something like `_store.Maintenance.SendAsync(new GetIndexesOperation(0, int.MaxValue), ct)` to get a list of all indexes.
Easy, works-

Now I want that for `myUrl/databases/myDatabase/indexes/progress` to see if there are indexes being built at the moment, but I can't find it in the documentation.

Any ideas?
Thanks!

**Comment by ayende** [comment]
Hi,
We don't expose that in the C# API, you can see the structure to add that here:
https://github.com/ravendb/ravendb/blob/c512fb46d50385be2250d39881157738c5bb9361/src/Raven.Client/Documents/Operations/Indexes/GetIndexesOperation.cs#L11

Can you explain what it is that you are trying to do with the progress endpoint? That is mostly for status & the stduio.

**Comment by stevencasburn** [comment]
Hi,
as you know we are indexing quite some data. After some data-imports the indexes need a while and our application won't run in that time. We want to make the progress visible in our applications so we don't think it's broken.

And as getting the indexes is within the C# API I thought it may make sense to also get the most obvious status about the indexes. :-)

**Comment by ayende** [comment, marked as answer]
Quick and dirty approach to getting this:

```
      using var context = JsonOperationContext.ShortTermSingleUse();
      var client = store.GetRequestExecutor(store.Database).HttpClient;
      var request = new HttpRequestMessage
      {
          Method = HttpMethod.Get,
          RequestUri = new Uri($"{store.Urls[0]}/databases/{store.Database}/indexes/progress")
      };
      var response = await client.SendAsync(request);
      response.EnsureSuccessStatusCode();
      var content = await response.Content.ReadAsStringAsync();
      var indexesProgress = JsonConvert.DeserializeObject<IndexesProgress>(content);

```

**Comment by stevencasburn** [comment]
Yeah, it is dirty, but I was planning to use it that way. Any chance you guys are integrating it into the API? Won't be a big deal, right? :-)

**Reply by ayende** [reply]
I added that to the issue tracker here: https://issues.hibernatingrhinos.com/issue/RavenDB-20324/Expose-GetIndexesProgress-in-the-client-API

**Reply by ayende** [reply]
See the full and complete solution there

**Reply by stevencasburn** [reply]
Thank you Dejan and you dev team

**Reply by stevencasburn** [reply]
It says:
State | Won't fix
Affected Release | 5.4.x

Which version is it gonna be? :-)

**Reply by ppekrol** [reply]
We will not apply it in the API, the code pasted in the issue can be used by you directly in a way that I wrote there. This endpoint and the classes there are internal and might subject to change. I suggest that you copy the code and use it directly in you application.

**Reply by stevencasburn** [reply]
Actually this doesn't work, as "DocumentConventions.**Default**" does not exist.
RavenDB.Client: v5.4.103

**Reply by ppekrol** [reply]
I forgot that `DocumentConventions.Default` are internal, but it is not hard to bypass it.

This is how the code looks.
```csharp
public static readonly DocumentConventions Default = new();
```
